zingolib/src/wallet/transaction_records_by_id.rs

```diff
@@ -20,9 +30,108 @@ impl std::ops::DerefMut for TransactionRecordsById {
         &mut self.0
     }
 }
+
+/// This block implements constructors.
+impl Default for TransactionRecordsById {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
 impl TransactionRecordsById {
+    // New empty Map
+    pub fn new() -> Self {
+        TransactionRecordsById(HashMap::new())
+    }
     // Associated function to create a TransactionRecordMap from a HashMap
     pub fn from_map(map: HashMap<TxId, TransactionRecord>) -> Self {
         TransactionRecordsById(map)
     }
 }
+/// This block implements methods to modify the map.
+impl TransactionRecordsById {
+    /// All information after a certain height is invalidated during a reorg.
+    pub fn invalidate_all_transactions_after_or_at_height(&mut self, reorg_height: BlockHeight) {
+        // First, collect txids that need to be removed
+        let txids_to_remove = self
+            .values()
+            .filter_map(|transaction_metadata| {
+                if transaction_metadata
+                    .status
+                    .is_confirmed_after_or_at(&reorg_height)
+                    || transaction_metadata
+                        .status
+                        .is_broadcast_after_or_at(&reorg_height)
+                // tODo: why dont we only remove confirmed transactions. unconfirmed transactions may still be valid in the mempool and may later confirm or expire.
+                {
+                    Some(transaction_metadata.txid)
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
+        self.invalidate_txids(txids_to_remove);
+    }
+    /// this function invalidiates a vec of txids by removing them and then all references to them.
+    pub(crate) fn invalidate_txids(&mut self, txids_to_remove: Vec<TxId>) {
```

Comment:
this invalidates a transaction but takes a txid right?
```suggestion
    pub(crate) fn invalidate_transactions(&mut self, txids_to_remove: Vec<TxId>) {
```

Reply:
true

Reply:
addressed